queries/ecma/highlights.scm

```diff
@@ -196,91 +234,99 @@
+(catch_clause body: (statement_block ["{" "}"] @punctuation.bracket.exception))
+(finally_clause body: (statement_block ["{" "}"] @punctuation.bracket.exception))
+
+(class_declaration body: (class_body ["{" "}"] @punctuation.bracket.class))
+
+(function_declaration parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(function_declaration parameters: (formal_parameters "," @punctuation.delimiter.function))
+(function_declaration body: (statement_block ["{" "}"] @punctuation.bracket.function))
+(generator_function_declaration parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(generator_function_declaration parameters: (formal_parameters "," @punctuation.delimiter.function))
+(generator_function_declaration body: (statement_block ["{" "}"] @punctuation.bracket.function))
+(function parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(function parameters: (formal_parameters "," @punctuation.delimiter.function))
+(function body: (statement_block ["{" "}"] @punctuation.bracket.function))
+(arrow_function parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(arrow_function parameters: (formal_parameters "," @punctuation.delimiter.function))
+(arrow_function "=>" @punctuation.special.function)
+(arrow_function body: (statement_block ["{" "}"] @punctuation.bracket.function))
+(generator_function_declaration "*" @punctuation.special.function)
+(method_definition "*" @punctuation.special.function)
+(method_definition parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(method_definition parameters: (formal_parameters "," @punctuation.delimiter.function))
+(method_definition body: (statement_block ["{" "}"] @punctuation.bracket.function))
 
 ; Keywords
 ;----------
 
-[
-  "if"
-  "else"
-  "switch"
-  "case"
-] @conditional
+["if" "else" "switch" "case" "default"] @keyword @conditional
 
-[
-  "import"
-  "from"
-] @include
+["import" "from" "as"] @keyword @include
 
-(export_specifier "as" @include)
-(import_specifier "as" @include)
-(namespace_export "as" @include)
-(namespace_import "as" @include)
+["new" "delete" "typeof" "in" "instanceof" "void"] @keyword @keyword.operator
 
-[
-  "for"
-  "of"
-  "do"
-  "while"
-  "continue"
-] @repeat
+["for" "do" "of" "while"] @keyword @repeat
 
-[
-  "break"
-  "class"
-  "const"
-  "debugger"
-  "export"
-  "extends"
-  "get"
-  "in"
-  "instanceof"
-  "let"
-  "set"
-  "static"
-  "target"
-  "typeof"
-  "var"
-  "with"
-] @keyword
+(for_in_statement operator: "in" @keyword @repeat)
 
-[
-  "async"
-  "await"
-] @keyword.coroutine
+["async" "await"] @keyword @keyword.async
 
 [
-  "return"
-  "yield"
-] @keyword.return
+"class"
+"extends"
+"static" @storageclass
+"get"
+"set"
+] @keyword @keyword.class
 
-[
-  "function"
-] @keyword.function
+"debugger" @keyword @keyword.debugger
 
-[
-  "new"
-  "delete"
-] @keyword.operator
+"with" @keyword @keyword.with
```

Comment:
Is there any specific reason why you would want both of these captures for one node?

Reply:
yep — I think it's useful to have ability for different highlights for different "types" of keywords

Reply:
As in, why are they captured with `@keyword` and `@keyword.x`. It will fall back to `@keyword` if `@keyword.x` doesn't exist, so no need to capture twice

Reply:
oh... that's nice! I didn't know about such "magic" ;-) will fix accordingly

Reply:
@lucario387 

> As in, why are they captured with `@keyword` and `@keyword.x`. It will fall back to `@keyword` if `@keyword.x` doesn't exist, so no need to capture twice

actually there is a case for double declaration `@x` and `@x.y` — if I have some highlights for both of them and wish to combine them accordingly — i.e. make all `@keyword` bold and have specific colours for each `@keyword.TYPE` (now it's applies only most precise highlight declaration, i.e. colours from `@keyword.TYPE` without anything from common `@keyword`)

Reply:
Then they can be applied from userspace instead. No need to flush the main queries with unnecessary separations like such